jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/AbstractUnassembledWebAppMojo.java

```diff
@@ -0,0 +1,252 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.maven.plugin;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.eclipse.jetty.util.resource.Resource;
+
+/**
+ * Base class for all goals that operate on unassembled webapps.
+ *
+ */
+public abstract class AbstractUnassembledWebAppMojo extends AbstractWebAppMojo
+{
+    /**
+     * The default location of the web.xml file. Will be used
+     * if &lt;webApp&gt;&lt;descriptor&gt; is not set.
+     */
+    @Parameter (defaultValue = "${project.baseDir}/src/main/webapp/WEB-INF/web.xml")
+    protected File webXml;
+    
+    /**
+     * The directory containing generated test classes.
+     * 
+     */
+    @Parameter (defaultValue = "${project.build.testOutputDirectory}", required = true)
+    protected File testClassesDirectory;
+    
+    /**
+     * An optional pattern for includes/excludes of classes in the testClassesDirectory
+     */
+    @Parameter
+    protected ScanPattern scanTestClassesPattern;
+
+    /**
+     * The directory containing generated classes.
+     */
+    @Parameter (defaultValue = "${project.build.outputDirectory}", required = true)
+    protected File classesDirectory;
+    
+
+    /**
+     * An optional pattern for includes/excludes of classes in the classesDirectory
+     */
+    @Parameter
```

Comment:
ok it work but it should be ` @Component ` from ` import org.apache.maven.plugins.annotations.Component; `